main/zoe/guide/contract-requirements.md

```diff
@@ -0,0 +1,59 @@
+# Contract Requirements
+
+<Zoe-Version/>
+
+When writing a smart contract that will be run on Zoe, it's important
+to know the proper format and other expectations. 
+
+A Zoe contract is a JavaScript file that can import other JavaScript
+code, including:
+
+* [@agoric/harden](https://www.npmjs.com/package/@agoric/harden): a package for recursively deep-freezing
+* [@agoric/nat](https://www.npmjs.com/package/@agoric/nat): a package
+  for testing whether something is a natural number (natural numbers
+  are recommended for currency-related programming in order to better
+  deal with rounding) and throwing if
+  not.
+* @agoric/notifier: A package that provides updates through
+  smartly resolving promises rather than polling
+* [@agoric/zoe](https://www.npmjs.com/package/@agoric/zoe): Zoe has
+  helpers that contracts can use by importing
+  `@agoric/zoe/src/contractSupport/zoeHelpers`
+
+A Zoe contract will be bundled up, so you should feel free to divide
+your contract into multiple files (perhaps putting helper functions in a
+separate file, for example) and import them.
+
+A Zoe contract needs to be able to run under [Agoric's SES](https://github.com/Agoric/ses-shim#secure-ecmascript-shim-ses-shim). Some legacy
+JavaScript code is incompatible with SES, because SES freezes the
```

Comment:
I'm not clear on what I should do with the info about the SES incompatibility. Is there a workaround or patch? The advice here seems to be "don't mutate JavaScript primordial objects" but what should I do instead?

Reply:
That's pretty much our answer: don't mutate them because it's bad programming practice. There's not a workaround or patch or anything like that, purposefully.